File: Python/Mahasiswa.py
from SivitasAkademik import SivitasAkademik


class Mahasiswa(SivitasAkademik):
    __nim = ""
    __fakultas = ""

    def __init__(self, nim="", fakultas="", nik="", nama="", jenis_kelamin="", asal_universitas="", email_edu=""):
        super().__init__(nik, nama, jenis_kelamin, asal_universitas, email_edu)
        self.__nim = nim
        self.__fakultas = fakultas

    def set_nim(self, nim):
        self.__nim = nim

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

    def get_nim(self):
        return self.__nim

    def get_fakultas(self):
        return self.__fakultas

File: CPP/Mahasiswa.cpp
class Mahasiswa : public SivitasAkademik
{
private:
    string nim;
    string fakultas;

public:
    Mahasiswa(string nim = "", string fakultas = "", string nik = "", string nama = "", string jenis_kelamin = "", string asal_universitas = "", string email_edu = "") : SivitasAkademik(nik, nama, jenis_kelamin, asal_universitas, email_edu)
    {
        this->nim = nim;
        this->fakultas = fakultas;
    }
    void set_nim(string nim)
    {
        this->nim = nim;
    }
    void set_fakultas(string fakultas)
    {
        this->fakultas = fakultas;
    }
    string get_nim()
    {
        return this->nim;
    }
    string get_fakultas()
    {
        return this->fakultas;
    }
};